android/README.md

```diff
@@ -0,0 +1,456 @@
+
+## Prerequisite
+
+- Supports Android applications written in Kotlin programming language.
+- The minimum supported SDK is API level 26, compiled to API level 34. However, there are certain limitations when using specific authentication methods:
+    - Passkeys are supported on API level 34 and above.
+    - Google authentication using the Credential Manager API is supported on API level 34 and above.
+
+## Getting Started
+
+### Register your application in /WSO2 Identity Server/Asgardeo
+
+#### WSO2 Identity Server
+
+1. Download the latest version of WSO2 Identity Server, and start the WSO2 Identity Server.
+2. Register a Mobile Application to integrate your application with WSO2 Identity Server. You will obtain a `client_ID` from WSO2 Identity Server for your application which will need to be embedded later for the SDK integration. Also note the redirect URI that you used to create the application, this is also required for the SDK integration.
+3. In the created mobile application go to the "Advanced" section and enable the application native authentication for your Android application.
+
+#### Asgardeo
+
+1. Register to Asgardeo and create an organization if you don't already have one. The organization name you choose will be referred to as `<org_name>` throughout this document.
+2. Register a Mobile Application in Asgardeo to integrate your application with Asgardeo. You will obtain a `client_ID` from Asgardeo for your application which will need to be embedded later for the SDK integration. Also note the redirect URI that you used to create the application, this is also required for the SDK integration.
+3. In the created mobile application go to the "Advanced" section and enable the application native authentication for your Android application.
+
+### Installing the SDK
+
+Add the latest released SDK in the `build.gradle` file of your Android application.
+
+```groovy
+dependencies {
+    implementation 'io.asgardeo.android:1.0.0'
+}
+```
+
+Add a redirect scheme in the Android application. You need to add the `appAuthRedirectScheme` in the application `build.gradle` file.
+
+This should be consistent with the CallBack Url of the Service Provider that you configured in the WSO2 Identity Server/Asgardeo.
+
+For example, if you have configured the `callBackUrl` as `wso2sample://oauth2`, then the `appAuthRedirectScheme` should be `wso2sample`.
+
+```groovy
+android.defaultConfig.manifestPlaceholders = [
+    'appAuthRedirectScheme': 'wso2sample'
+]
+```
+
+### Start the authentication process
+
+1. First, you need to initialize the SDK object, `AsgardeoAuth`, to authenticate users into your application. This can be done in a repository if you are using an MVVM pattern in your application.
+
+```kotlin
+private val asgardeoAuth: AsgardeoAuth = AsgardeoAuth.getInstance(
+    AuthenticationCoreConfig(
+        discoveryEndpoint = "https://localhost:9443//oauth2/token/.well-known/openid-configuration",
```

Comment:
Will devs need to specify the discovery endpoint by themselves, when initializing sdk? I doubt that would be good DX. WDYT?

```suggestion
        discoveryEndpoint = "https://localhost:9443/oauth2/token/.well-known/openid-configuration",
```